File: cpp/examples/test_dds.cpp
#include "unitree/dds_wrapper/go2/go2_sub.h"
#include "unitree/dds_wrapper/go2/go2_pub.h"

int main(int argc, char** argv)
{
    // Init DDS
    std::string iface = (argc > 1) ? argv[1] : "";
    unitree::robot::ChannelFactory::Instance()->Init(0, iface);

    auto pub = std::make_unique<unitree::robot::go2::publisher::LowState>();
    auto sub = std::make_shared<unitree::robot::go2::subscription::LowState>();

    while (true)
    {
        pub->msg_.crc() = pub->msg_.crc() + 1;
        pub->unlockAndPublish();

        sleep(1);

        std::cout << "pub->msg_.crc(): " << pub->msg_.crc() << std::endl;
        std::cout << "sub->msg_.crc(): " << sub->msg_.crc() << std::endl;
    }

    return 0;
}

File: python/examples/test_dds.py
from unitree_dds_wrapper.publisher import Publisher
from unitree_dds_wrapper.subscription import Subscription
from unitree_dds_wrapper.idl import unitree_go
import time

msg_type = unitree_go.msg.dds_.LowState_
pub = Publisher(message=msg_type, topic="rt/test_dds")
sub = Subscription(message=msg_type, topic="rt/test_dds")

while True:
  pub.msg.level_flag = 1 - pub.msg.level_flag
  pub.write()
  time.sleep(0.5)

  print("send: ", pub.msg.level_flag)

  if sub.msg:
    print("receive: ", sub.msg.level_flag)